frontend/components/SetupPage/Create/BridgeInProgress.tsx

```diff
@@ -100,54 +118,61 @@ const useBridgingSteps = (quoteId: string) => {
         }),
       };
     },
+    // refetch every 5 seconds until the status is finished
+    refetchInterval: ({ state }) => {
+      const status = state?.data?.status;
+      if (status === 'FINISHED') return false;
+      return FIVE_SECONDS_INTERVAL;
+    },
     enabled: !!quoteId && isOnline && !!bridgeExecute,
   });
 };
 
-// TODO: to update
-const useMasterSafeCreation = () => ({
-  isLoading: false,
-  isError: false,
-  data: {
-    isSafeCreated: false,
-    txnLink:
-      'https://etherscan.io/tx/0x3795206347eae1537d852bea05e36c3e76b08cefdfa2d772e24bac2e24f31db3',
-  },
-});
-
-// TODO: to update
-const useMasterSafeTransfers = () => ({
-  isLoading: false,
-  isError: false,
-  data: {
-    status: 'CREATED',
-    transfers: [
-      {
-        symbol: 'OLAS' as TokenSymbol,
-        status: 'wait' as BridgingStepStatus,
-        txnLink: null,
-      },
-      {
-        symbol: 'OLAS' as TokenSymbol,
-        status: 'wait' as BridgingStepStatus,
-        txnLink: null,
-      },
-    ],
-  },
-});
+const useMasterSafeCreation = () => {
+  const backupSignerAddress = useBackupSigner();
+  const symbols = useSymbols();
+  const { masterEoa } = useMasterWalletContext();
+  const { balances, isBalancesAndFundingRequirementsLoading } =
+    useBalanceAndRefillRequirementsContext();
+  const { selectedAgentConfig } = useServices();
 
-const useTimeRemaining = () => {
-  const TIME_FOR_SAFE_CREATION = ONE_SECOND_INTERVAL * 4; // TODO: to update
-  const TIME_FOR_MASTER_SAFE_TRANSFER = ONE_SECOND_INTERVAL * 4; // TODO: to update
-  const timeToExecuteQuote = 1744690251; // TODO: to update
-  return {
-    isLoading: false,
-    isError: false,
-    timeRemaining:
-      timeToExecuteQuote +
-      TIME_FOR_SAFE_CREATION +
-      TIME_FOR_MASTER_SAFE_TRANSFER,
-  };
+  const chain = selectedAgentConfig.middlewareHomeChainId;
+
+  const initialFunds = useMemo(() => {
+    if (isBalancesAndFundingRequirementsLoading) return;
+    if (!balances) return;
+    if (!masterEoa) return;
+
+    const eoaFunds = { ...balances[masterEoa.address] };
+    const fundRequiredByEoa = 50; // Doubt: where is this in refill_requirements? 🥲
```

Comment:
afair the values are hardcoded in MW 🫠 and we only know from the refill requirements endpoint when the balance is zero

Reply:
worth asking Jose